src/scenes/Projects/Members/Update/UpdateProjectMember.tsx

```diff
@@ -0,0 +1,117 @@
+import { Container } from '@material-ui/core';
+import { Skeleton } from '@material-ui/lab';
+import React from 'react';
+import { Except } from 'type-fest';
+import { displayRole, Role, RoleList } from '../../../../api';
+import {
+  DialogForm,
+  DialogFormProps,
+} from '../../../../components/Dialog/DialogForm';
+import { SubmitButton, SubmitError } from '../../../../components/form';
+import { AutocompleteField } from '../../../../components/form/AutocompleteField';
+import { ProjectMembersDocument } from '../List/ProjectMembers.generated';
+import {
+  useDeleteProjectMemberMutation,
+  useGetUserRolesQuery,
+  useUpdateProjectMemberMutation,
+} from './UpdateProjectMember.generated';
+
+interface DialogValues {
+  projectMember: {
+    id: string;
+    roles?: readonly Role[];
+  };
+}
+
+type UpdateProjectMemberProps = Except<
+  DialogFormProps<DialogValues>,
+  'onSubmit' | 'initialValues'
+> & {
+  id: string;
+  userId: string;
+  userRoles?: readonly Role[];
+  projectId: string;
+};
```

Comment:
```suggestion

import { UpdateProjectMemberInput } from '../../../../api';

export interface UpdateProjectMemberFormParams {
  projectMember: UpdateProjectMemberInput['projectMember'];
  userId: string;
}

type UpdateProjectMemberProps = Except<
  DialogFormProps<UpdateProjectMemberInput>,
  'onSubmit' | 'initialValues'
> &
  UpdateProjectMemberFormParams;
```

It's safer to base these values on the actual input type the mutation will be expecting (`UpdateProjectMemberInput`). This way if it ever changes we don't have to rewrite these typings. 

Also, you can then import `UpdateProjectMemberFormParams` in `ProjectMembersList` and pass it as the type variable to `useDialog`.

Reply:
Good idea, done